app/src/main/java/io/gnosis/safe/helpers/ConnectivityInfoProvider.kt

```diff
@@ -0,0 +1,42 @@
+package io.gnosis.safe.helpers
+
+import android.net.ConnectivityManager
+import android.net.Network
+import android.net.NetworkCapabilities
+import android.net.NetworkRequest
+import java.io.IOException
+
+class ConnectivityInfoProvider(private val connectivityManager: ConnectivityManager) {
+
+    var offline: Boolean = true
+
+    private val networkCallback = object : ConnectivityManager.NetworkCallback() {
+        override fun onAvailable(network: Network?) {
+            super.onAvailable(network)
+            offline = false
+        }
+
+        override fun onLost(network: Network?) {
+            super.onLost(network)
```

Comment:
So if I have wifi and cellular and then wifi disconnects ... what happens here?

Reply:
fixed it - offline flag is set only if there is no network available

Reply:
It is a problem. If either disconnect, the state goes to offline. Which is false as the phone can use the other network.

Reply:
Actually when I tried this again, it is ok. We lose wifi but gain cellular a moment later.